TypeCobol/Compiler/Diagnostics/CrossChecker.cs

```diff
@@ -984,57 +971,53 @@ static class SectionOrParagraphUsageChecker
         /// Disambiguate between Section or Paragraph reference.
         /// </summary>
         /// <param name="callerNode">Node using the paragraph or the reference.</param>
-        /// <param name="target">A non-null Symbol reference to disambiguate.</param>
+        /// <param name="target">A Symbol reference to disambiguate.</param>
         /// <param name="currentSection">The node scope in which the perform statement is declared</param>
-        /// <returns>A tuple made of a SymbolType and a Node when the target has been correctly resolved.
-        /// The returned SymbolType is non-ambiguous if the type of the target has been determined.</returns>
+        /// <returns>A tuple corresponding to the paragraph OR section found.
+        /// The returned tuple has one null value and one filled value if the reference is non-ambiguous.</returns>
         /// <remarks>This method will create appropriate diagnostics on Node if resolution is inconclusive.</remarks>
-        public static (SymbolType, Node) ResolveSectionOrParagraph(Node callerNode, [NotNull] SymbolReference target, Section currentSection)
+        public static (ParagraphSymbol, SectionSymbol) ResolveSectionOrParagraph(Node callerNode, SymbolReference target, Section currentSection)
```

Comment:
It would be more intuitive if the tuple items are in the same order as the method name, so either change the order of the tuple elements or rename the method.

Reply:
Renaming the method was easier :wink:
See https://github.com/TypeCobolTeam/TypeCobol/pull/2003/commits/a488959c6e0dba2b6fd649e0181a941167802a9b